src/mono/mono/metadata/method-builder-ilgen.c

```diff
@@ -652,3 +700,73 @@ mono_mb_set_param_names (MonoMethodBuilder *mb, const char **param_names)
 {
 	mb->param_names = param_names;
 }
+
+void
+mono_mb_set_wrapper_data_kind (MonoMethodBuilder *mb, uint16_t wrapper_data_kind)
+{
+	g_assert (mb->inflate_wrapper_data);
+	MonoMethodWrapper *mw = (MonoMethodWrapper*)mb->method;
+	// index of the data added by most recent mono_mb_add_data
+	int idx = g_list_length((GList *)mw->method_data);
+	g_assert (idx > 0 && idx <= UINT16_MAX);
+
+	MonoMethodBuilderInflateWrapperData *info = g_new (MonoMethodBuilderInflateWrapperData, 1);
+	info->idx = (uint16_t)idx;
+	info->kind = wrapper_data_kind;
+	mb->wrapper_data_inflate_info = g_list_prepend (mb->wrapper_data_inflate_info, info);
+}
+
+gboolean
+mono_mb_inflate_generic_wrapper_data (MonoGenericContext *context, gpointer *method_data, MonoError *error)
+{
+	MonoMethodBuilderInflateWrapperData* inflate_info = (MonoMethodBuilderInflateWrapperData*)method_data[MONO_MB_ILGEN_INFLATE_WRAPPER_INFO_IDX];
+	MonoMethodBuilderInflateWrapperData* p = inflate_info;
+	int count = 0;
+	while (p->idx != 0) {p++; count++;}
```

Comment:
Is this necessary?

Reply:
I am wondering why it can't be done in the while loop?

Reply:
oh. I see what you mean.  Yea good point